src/main/java/com/google/cloud/teleport/splunk/SplunkEventWriter.java

```diff
@@ -290,6 +295,18 @@ private static void flushWriteFailures(
       receiver.output(error);
     }
   }
+  
+  /**
+   * Checks whether the HEC URL matches the format 'http://domain' or 'https://domain'.
+   *
+   * @param url for HEC event collector
+   * @return true if the URL is valid
+   */
+  private static boolean isValidUrlFormat(String url) {
+    Pattern pattern = Pattern.compile("^http(s?)://.+$");
```

Comment:
Good to have this new function `isValidUrlFormat`.
However, incorrect URLs like https://example or https://1.2.3 will be marked as valid here.

I recommend restricting the regex to ensure correct URL, including either a FQDN or IP address.

Reply:
Why not use a validator library instead of attempting to rewrite the regex and parsing logic? Something like Apache Commons Validator? There are a number of permutations in verifying the validity of a URL and the validator library covers most of those.

Reply:
Restricted the regex to allow URLs that match the format [PROTOCOL]://[HOST]:[PORT]. This would also take care of the [issue](https://stackoverflow.com/questions/64866113/pub-sub-to-splunk-dataflow-template-the-requested-url-was-not-found-on-this-se) where `services/collector/event` is appended to the URL.

Also added the `InetAddresses` and `InternetDomainName` utilities from Guava to validate the host

Reply:
The port is optional. I think this will invalidate any URL without a port.
Also a path in the URL is completely valid: it's possible there's load balancer at the destination performing some path-based routing before it goes to HEC.

Reply:
Template already internally appends path fragment(services/collector/event) to user provided URL. https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/788f6b7af095d3455d8c8f5da18c320baeeaab71/src/main/java/com/google/cloud/teleport/splunk/HttpEventPublisher.java#L75 

It will bit tricky to handle all scenarios. For e.g: even using apache commons urlvalidator is also not supporting all our required scenarios (e.g: not failing when path fragment is included)

```
 boolean isValidUrlFormat(String url) {
        String[] schemes = {"http", "https"};
        UrlValidator urlValidator =   new UrlValidator(schemes,  UrlValidator.ALLOW_LOCAL_URLS + UrlValidator.NO_FRAGMENTS);
        return urlValidator.isValid(url);
    }

 for(String url : new String[]{"http://testurl"
                ,"https://example.splunk.com:1234"
                ,"https://10.209.127.123:8088",
                "https://example.splunk.com:1234/services/collector/event"}) {
            System.out.printf("Url: %s - isValid: %s\n", url, isValidUrlFormat(url));
        }
```

We would like the users to stick to a specific format as per the documentation https://cloud.google.com/dataflow/docs/guides/templates/provided-streaming#cloudpubsubtosplunk.  
https://docs.splunk.com/Documentation/Splunk/8.1.2/Data/UsetheHTTPEventCollector (<protocol>://<host>:<port>)

If we need to make port optional then the regex can be updated to ```"^http(s?)://([^\\:]+)(:[0-9]+)?$"``` with enough testing

Reply:
Updated the regex to make the port optional. Also added a check to invalidate the URL if the host has `services/collector/event` appended to it.